### PR TITLE
Smit/add friend

### PR DESCRIPTION
**Pull request by tailorsmit**
add friends and friend list

### CODE REVIEW

File: springbackend/src/main/java/com/simplecoding/social/service/FriendService.java [hunk 1]
```diff
@@ -40,22 +40,34 @@ public void saveFriend(UserDto userDto1, int id) throws NullPointerException{
...
-
+        User firstuser = user1;
+        User seconduser = user2;
+        if(user1.getId() > user2.getId()){
+             firstuser = user2;
+             seconduser = user1;
+        }
+        if( !(friendRepository.existsByFirstUserAndSecondUser(firstuser,seconduser)) ){
+            friend.setFirstUser(firstuser);
+            friend.setSecondUser(seconduser);
```
Comment (nilmadhab) [inline springbackend/src/main/java/com/simplecoding/social/service/FriendService.java:51]: add date field

File: springbackend/src/main/java/com/simplecoding/social/service/FriendService.java [hunk 2]
```diff
@@ -40,22 +40,34 @@ public void saveFriend(UserDto userDto1, int id) throws NullPointerException{
...
             friendRepository.save(friend);
-            friendRepository.save(friendViceVersa);
         }
+//        Friend friendViceVersa = new Friend();
+//        friendViceVersa.setCreatedDate(new Date());
+//        User firstUser = userRepository.findUserByEmail(userDto2.getEmail());
+//        User secondUser = userRepository.findUserByEmail(userDto1.getEmail());
+//
```
Comment (nilmadhab) [inline springbackend/src/main/java/com/simplecoding/social/service/FriendService.java]: remove the comments

File: springbackend/src/main/java/com/simplecoding/social/service/FriendService.java [hunk 3]
```diff
@@ -64,11 +76,15 @@ public void saveFriend(UserDto userDto1, int id) throws NullPointerException{
         UserDto currentUserDto = securityService.getUser();
         User currentUser = userRepository.findUserByEmail(currentUserDto.getEmail());
         List<Friend> friends = friendRepository.findByFirstUser(currentUser);
```
Comment (nilmadhab) [inline springbackend/src/main/java/com/simplecoding/social/service/FriendService.java]: if you lexiographically adding when do you need to get friends by both first and second user?

File: springbackend/src/main/java/com/simplecoding/social/service/FriendService.java [hunk 4]
```diff
@@ -64,11 +76,15 @@ public void saveFriend(UserDto userDto1, int id) throws NullPointerException{
...
-        List<User> friendUsers = new ArrayList<User>();
+        List<Friend> friends1 = friendRepository.findBySecondUser(currentUser);
```
Comment (nilmadhab) [inline springbackend/src/main/java/com/simplecoding/social/service/FriendService.java]: if you lexiographically adding when do you need to get friends by both first and second user?

File: vue-social/src/router/index.js [hunk 5]
```diff
@@ -7,6 +7,7 @@ import UserList from '../views/ListUsers.vue'
 import Chat from '../components/Chat.vue'
 import Room from '../components/Room.vue'
 import AddRoom from '../components/AddRoom.vue'
+import listFriends from "../views/listFriends";
```
Comment (nilmadhab) [inline vue-social/src/router/index.js]: make it ListFiends,( first letter capital like other imports)

File: vue-social/src/router/index.js [hunk 6]
```diff
@@ -7,6 +7,7 @@ import UserList from '../views/ListUsers.vue'
 import Chat from '../components/Chat.vue'
 import Room from '../components/Room.vue'
 import AddRoom from '../components/AddRoom.vue'
+import listFriends from "../views/ListFriends";
```
Comment (nilmadhab) [inline vue-social/src/router/index.js]: change listFriends to ListFriends, follow the code convention

File: vue-social/src/router/index.js [hunk 7]
```diff
@@ -29,6 +30,10 @@ const routes = [
     path: '/listusers',
     name: 'ListUsers',
     component: UserList
+  },{
+    path: '/listfriends',
+    name: 'ListFriends',
+    component: listFriends
```
Comment (nilmadhab) [inline vue-social/src/router/index.js]: make it ListFriends from listFriends

File: springbackend/src/main/java/com/simplecoding/social/service/FriendService.java (hunk 1 above)
Comment (nilmadhab) [reply springbackend/src/main/java/com/simplecoding/social/service/FriendService.java:51]: you did not fix the issue but marked as resolve

Comment (tailorsmit) [reply springbackend/src/main/java/com/simplecoding/social/service/FriendService.java:51]: Date was already added in above line no 40

File: vue-social/src/router/index.js (hunk 7 above)
Comment (tailorsmit) [reply vue-social/src/router/index.js]: changes applied.

File: vue-social/src/router/index.js (hunk 6 above)
Comment (tailorsmit) [reply vue-social/src/router/index.js]: changed it.

**Review by nilmadhab — APPROVED** [review]

**Pull request merged**